### PR TITLE
Fix panic while importing MFA Duo resource

### LINKED ISSUE #1606: Panic importing `vault_mfa_duo`
### Terraform Version
```
Terraform v1.2.8
on linux_amd64
+ provider registry.terraform.io/hashicorp/vault v2.24.1
```

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_mfa_duo

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "vault_mfa_duo" "duo" {
  name            = "duo"
  mount_accessor  = vault_ldap_auth_backend.ldap.accessor
  secret_key      = "xxx"
  integration_key = "xxx"
  api_hostname    = "api-2b5c39f5.duosecurity.com"
}
```

### Panic Output

```
❯ terraform import vault_mfa_duo.duo duo
vault_mfa_duo.duo: Importing from ID "duo"...
vault_mfa_duo.duo: Import prepared!
  Prepared vault_mfa_duo for import
vault_mfa_duo.duo: Refreshing state... [id=duo]
╷
│ Error: Request cancelled
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵


Stack trace from the terraform-provider-vault_v2.24.1_x4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xf1e154]

goroutine 14 [running]:
github.com/hashicorp/terraform-provider-vault/vault.mfaDuoRead(0xc00088e8c0, 0x119aa00, 0xc0003b8280, 0xc00088e8c0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_mfa_duo.go:115 +0x2f4
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00056def0, 0xc000740c30, 0x119aa00, 0xc0003b8280, 0xc0007e6238, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.0/helper/schema/resource.go:460 +0x12e
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000286c30, 0x13c3450, 0xc0008913b0, 0xc0007ca360, 0xc000286c30, 0xc0008913b0, 0xc0005d1ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.0/internal/helper/plugin/grpc_provider.go:525 +0x3dd
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x1168220, 0xc000286c30, 0x13c3450, 0xc0008913b0, 0xc0007ca300, 0x0, 0x13c3450, 0xc0008913b0, 0xc00047c6c0, 0x90)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.0/internal/tfplugin5/tfplugin5.pb.go:3269 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000360000, 0x13ce238, 0xc000103980, 0xc00092e100, 0xc000692330, 0x19de4b0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1082 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000360000, 0x13ce238, 0xc000103980, 0xc00092e100, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1405 +0xccf
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000345b0, 0xc000360000, 0x13ce238, 0xc000103980, 0xc00092e100)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:746 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:744 +0xa5

Error: The terraform-provider-vault_v2.24.1_x4 plugin crashed!
```

### PR DESCRIPTION
**Pull request by vinay-gopalan**
Relates OR Closes #1606 #1527 

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestMFADuoBasic'
=== RUN   TestMFADuoBasic
--- PASS: TestMFADuoBasic (1.42s)
PASS
```

### CODE REVIEW

File: vault/resource_mfa_duo.go [hunk 1]
```diff
@@ -81,55 +85,60 @@ func mfaDuoWrite(d *schema.ResourceData, meta interface{}) error {
...
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
-	name := d.Get("name").(string)
+	name := d.Id()
 
 	resp, err := client.Logical().Read(mfaDuoPath(name))
```
Comment (benashz) [inline vault/resource_mfa_duo.go:120]: We should probably be handling the case where `resp` is `nil`. In that case we would return nil to denote that resource does not exist in Vault.

**Review by benashz — COMMENTED** [review]
Looking good. I noticed a small issue in  `mfaDuoRead()`. We should probably address that before merging.

File: vault/resource_mfa_duo.go (hunk 1 above)
Comment (vinay-gopalan) [reply vault/resource_mfa_duo.go:120]: Good call! Added this check on [L121](https://github.com/hashicorp/terraform-provider-vault/pull/1669/files#diff-90462eca7e943acf538bf9f4b84b87b5588cde9d4cde65947b87c8f1e24683d7R121)

**Review by benashz — APPROVED** [review]
LGTM!

**Pull request merged**